addons/purchase_stock/models/stock_rule.py

```diff
@@ -141,16 +141,17 @@ def _run_buy(self, procurements):
                         procurement.values, po))
             self.env['purchase.order.line'].sudo().create(po_line_values)
 
-    def _get_lead_days(self, product):
+    def _get_lead_days(self, product, **values):
         """Add the company security lead time, days to purchase and the supplier
         delay to the cumulative delay and cumulative description. The days to
         purchase and company lead time are always displayed for onboarding
         purpose in order to indicate that those options are available.
         """
-        delay, delay_description = super()._get_lead_days(product)
+        delay, delay_description = super()._get_lead_days(product, **values)
         bypass_delay_description = self.env.context.get('bypass_delay_description')
         buy_rule = self.filtered(lambda r: r.action == 'buy')
-        seller = product.with_company(buy_rule.company_id)._select_seller()
+        seller = values.get('supplierinfo', product.with_company(
```

Comment:
Prefer a ternary operator to avoid a extra call to `_select_seller` (or a old fashion `if`)
```python
seller = values.get('supplierinfo') if 'supplierinfo' in values else product.with_company(buy_rule.company_id)._select_seller()
```